code/game/objects/items/storage/storage.dm

```diff
@@ -383,69 +384,94 @@
+	if(!access_delay || !should_access_delay(accessed, user, taking_out))
+		return TRUE
+
+	if(!alert_user)
+		return do_after(user, access_delay, TRUE, src, ignore_turf_checks=TRUE)
+
+	to_chat(user, "<span class='notice'>You begin to [taking_out ? "take" : "put"] [accessed] [taking_out ? "out of" : "into"] [src]")
+	if(!do_after(user, access_delay, TRUE, src, ignore_turf_checks=TRUE))
+		to_chat(user, "<span class='warning'>You fumble [accessed]!</span>")
+		return FALSE
+	return TRUE
+
+//This proc checks to see if we should actually delay access in this scenario.
+//This proc should return TRUE or FALSE
+/obj/item/storage/proc/should_access_delay(obj/item/accessed, mob/user, taking_out)
+	return FALSE
+
 //This proc handles items being inserted. It does not perform any checks of whether an item can or can't be inserted. That's done by can_be_inserted()
 //The stop_warning parameter will stop the insertion message from being displayed. It is intended for cases where you are inserting multiple items at once,
 //such as when picking up all the items on a tile with one click.
 //user can be null, it refers to the potential mob doing the insertion.
-/obj/item/storage/proc/handle_item_insertion(obj/item/W, prevent_warning = 0, mob/user)
-	if(!istype(W)) return 0
-	if(user && W.loc == user)
-		if(!user.transferItemToLoc(W, src))
-			return 0
+/obj/item/storage/proc/handle_item_insertion(obj/item/item, prevent_warning = 0, mob/user)
+	if(!istype(item)) return FALSE
+	if(!handle_access_delay(item, user, taking_out=FALSE))
+		item.forceMove(item.drop_location())
+		return
+	if(user && item.loc == user)
+		if(!user.transferItemToLoc(item, src))
+			return FALSE
 	else
-		W.forceMove(src)
-	W.on_enter_storage(src)
+		item.forceMove(src)
+	item.on_enter_storage(src)
 	if(user)
 		if (user.client && user.s_active != src)
-			user.client.screen -= W
+			user.client.screen -= item
 		if(!prevent_warning)
-			var/visidist = W.w_class >= 3 ? 3 : 1
-			user.visible_message("<span class='notice'>[usr] puts [W] into [src].</span>",\
-								"<span class='notice'>You put \the [W] into [src].</span>",\
+			var/visidist = item.w_class >= 3 ? 3 : 1
+			user.visible_message("<span class='notice'>[usr] puts [item] into [src].</span>",\
+								"<span class='notice'>You put \the [item] into [src].</span>",\
 								null, visidist)
 	orient2hud()
 	for(var/mob/M in can_see_content())
 		show_to(M)
 	if (storage_slots)
-		W.mouse_opacity = 2 //not having to click the item's tiny sprite to take it out of the storage.
+		item.mouse_opacity = 2 //not having to click the item's tiny sprite to take it out of the storage.
 	update_icon()
 	return 1
 
 //Call this proc to handle the removal of an item from the storage item. The item will be moved to the atom sent as new_target
-/obj/item/storage/proc/remove_from_storage(obj/item/I, atom/new_location)
-	if(!istype(I))
+/obj/item/storage/proc/remove_from_storage(obj/item/item, atom/new_location, mob/user)
```

Comment:
you should arguably not change the names of the var you're changing gitblame for barely reason

Reply:
It's always better to have names follow the guidelines.
And gitblame is smart enough to track this.